gemma/ops.h

```diff
@@ -113,12 +150,38 @@ HWY_INLINE void MatVecAddLoop(const ArrayT& mat, const size_t mat_ofs,
   }
 }
 
+template <bool kAdd, size_t kOuter, size_t kInner, typename VecT, typename AddT,
+          size_t kCapacity>
+HWY_INLINE void MatVecAddLoop(
+    const CompressedArray<hwy::bfloat16_t, kCapacity>& mat,
+    const size_t mat_ofs,
+    const VecT* HWY_RESTRICT vec_aligned,
+    const AddT* HWY_RESTRICT add,
+    float* HWY_RESTRICT out) {
+  PROFILER_ZONE("MatVecAddLoop");
+
+  const hn::ScalableTag<float> df;
+
+  const auto vec_dequant = hwy::AllocateAligned<float>(kInner);
```

Comment:
Allocation can be quite slow, let's move this into gemma.cc's Activations. That would require plumbing through an extra tmp arg here, and the std::array storage should probably be the largest per-call size * max number of threads (say 128 or 256). Would you prefer if I made this change?

Reply:
Done in #173, see `even_odd` :)